lambdajs/src/SharedAbstract.hs

```diff
@@ -0,0 +1,262 @@
+{-# LANGUAGE Arrows                     #-}
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE FlexibleContexts           #-}
+{-# LANGUAGE FlexibleInstances          #-}
+{-# LANGUAGE FunctionalDependencies     #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses      #-}
+{-# LANGUAGE StandaloneDeriving         #-}
+{-# LANGUAGE TypeSynonymInstances       #-}
+module SharedAbstract where
+
+import           Derivations
+import           GHC.Generics                                    (Generic)
+import           Prelude                                         hiding (break,
+                                                                  error, fail,
+                                                                  lookup, map,
+                                                                  read)
+import qualified Prelude
+import           SharedInterpreter
+import           Syntax
+
+import           Data.Hashable
+import           Data.Identifiable
+import           Data.Set
+
+import           Data.Abstract.Environment
+import           Data.Abstract.HandleError
+import           Data.Abstract.Store
+import           Data.List                                       (find)
+import           Data.Order
+
+import           Control.Arrow.Transformer.Abstract.Environment
+import           Control.Arrow.Transformer.Abstract.HandleExcept
+import           Control.Arrow.Transformer.Abstract.Store
+import           Control.Arrow.Transformer.State
+import           Control.Arrow.Utils                             (map, pi1, pi2)
+
+import           Control.Arrow
+import           Control.Arrow.Environment
+import           Control.Arrow.Except
+import           Control.Arrow.Fail
+import           Control.Arrow.Reader
+import           Control.Arrow.State
+import           Control.Arrow.Store
+import           Control.Category
+
+
+instance PreOrd Type where
```

Comment:
If you use powersets of types you should not need this instance: https://github.com/svenkeidel/sturdy/blob/master/lib/src/Data/Abstract/DiscretePowerset.hs#L32